pennylane_lightning/core/src/simulators/lightning_tensor/tncuda/TNCudaBase.hpp

```diff
@@ -60,133 +57,197 @@ class TNCudaBase : public TensornetBase<PrecisionT, Derived> {
+        get_state_tensor(res);
     }
 
     /**
-     * @brief Get the cutensornet handle that the object is using.
+     * @brief Get the full state vector representation of a quantum state.
      *
-     * @return cutensornetHandle_t
+     * @return std::vector<ComplexT> Full state vector representation of the
+     * quantum state on host
      */
-    [[nodiscard]] auto getTNCudaHandle() const -> cutensornetHandle_t {
-        return handle_.get();
+    auto getDataVector() -> std::vector<ComplexT> {
+        std::size_t length = std::size_t{1} << BaseType::getNumQubits();
+        std::vector<ComplexT> results(length);
+
+        getData(results.data(), results.size());
+
+        return results;
     }
 
     /**
-     * @brief Access the CublasCaller the object is using.
-     *
-     * @return a reference to the object's CublasCaller object.
+     * @brief Set current quantum state as zero state.
      */
-    auto getCublasCaller() const -> const CublasCaller & {
-        return *cublascaller_;
+    void reset() {
+        const std::vector<std::size_t> zeroState(BaseType::getNumQubits(), 0);
+        setBasisState(zeroState);
     }
 
     /**
-     * @brief Get the quantum state pointer.
-     *
-     * @return cutensornetState_t
+     * @brief Update quantum state with a basis state.
+     * NOTE: This API assumes the bond vector is a standard basis vector
+     * ([1,0,0,......]) and current implementation only works for qubit systems.
+     * @param basisState Vector representation of a basis state.
      */
-    [[nodiscard]] auto getQuantumState() const -> cutensornetState_t {
-        return quantumState_;
+    void setBasisState(const std::vector<std::size_t> &basisState) {
+        PL_ABORT_IF(BaseType::getNumQubits() != basisState.size(),
+                    "The size of a basis state should be equal to the number "
+                    "of qubits.");
+
+        bool allZeroOrOne = std::all_of(
+            basisState.begin(), basisState.end(),
+            [](std::size_t bitVal) { return bitVal == 0 || bitVal == 1; });
+
+        PL_ABORT_IF_NOT(allZeroOrOne,
+                        "Please ensure all elements of a basis state should be "
+                        "either 0 or 1.");
+
+        CFP_t value_cu = cuUtil::complexToCu<ComplexT>(ComplexT{1.0, 0.0});
+
+        for (std::size_t i = 0; i < BaseType::getNumQubits(); i++) {
+            tensors_[i].getDataBuffer().zeroInit();
+            std::size_t target = 0;
+            std::size_t idx = BaseType::getNumQubits() - std::size_t{1} - i;
+
+            // Rightmost site
+
+            if (i == 0) {
+                target = basisState[idx];
+            } else {
+                target = basisState[idx] == 0 ? 0 : bondDims_[i - 1];
+            }
+
+            PL_CUDA_IS_SUCCESS(
+                cudaMemcpy(&tensors_[i].getDataBuffer().getData()[target],
+                           &value_cu, sizeof(CFP_t), cudaMemcpyHostToDevice));
+        }
```

Comment:
Could we cuda copy the 0-qubit test prior to this for-loop and start the loop at `i = 1` to simplify the logic here a bit?